config/eslint.js

```diff
@@ -46,7 +57,32 @@ module.exports = {
     '@typescript-eslint/no-redeclare': ['error'],
     '@typescript-eslint/restrict-plus-operands': 'off',
     'import/no-default-export' : ['error'],
-    '@typescript-eslint/strict-boolean-expressions': ['error']
+    '@typescript-eslint/strict-boolean-expressions': ['error'],
+    'simple-import-sort/imports': 'error',
+    'simple-import-sort/exports': 'error',
+    'import/default': 'error',
+    'import/export': 'error',
+    'import/exports-last': 'off', // TODO: set to `warn` for fixing and then `error`
+    'import/extensions': 'off',
+    'import/first': 'error',
+    'import/group-exports': 'off',
+    'import/namespace': 'error',
+    'import/no-absolute-path': 'error',
+    'import/no-anonymous-default-export': 'error',
+    'import/no-cycle': 'off', // TODO: set to `warn` for fixing and then `error`
+    'import/no-deprecated': 'off', // TODO: set to `warn` for fixing and then `error`
+    'import/no-duplicates': 'error',
+    'import/no-dynamic-require': 'off',
+    'import/no-extraneous-dependencies': 'error',
+    'import/no-mutable-exports': 'error',
+    'import/no-namespace': 'off',
+    'import/no-restricted-paths': 'error',
+    'import/no-self-import': 'error',
+    'import/no-unresolved': 'off',
+    'import/no-unused-modules': 'error',
+    'import/no-useless-path-segments': 'error',
+    'import/no-webpack-loader-syntax': 'error',
+    'import/order': 'error',
```

Comment:
Do we actually need all those additional rules? My intuition would be to only include rules that currently result in an update when applied for performance and general simplicity. I haven't looked up what each of these rules does in detail but here's a couple examples of rules that I suspect don't result in changes:
```
    'import/no-absolute-path': 'error',
    'import/no-anonymous-default-export': 'error',
    'import/no-restricted-paths': 'error',
    'import/no-self-import': 'error',
    
    Curious to hear your thoughts on that, and more generally, I personally feel like it'd be great if the additional rule was justified, so that we avoid bloating up our linting configuration.

Reply:
Yeah I can totally get behind the preventative approach, I guess my main concern is I'd like to make sure that the added rules are relevant, mostly because I don't currently have a clear view on why some of these rules were added to start with.

I think it's worth thinking about what our criteria for adding new rules should be. Some examples that come to mind:

-  we are adding `'import/default'` (which ensures that default imports are coupled with exports) but not `'import/named'` (which ensures that named imports are coupled with exports). 
- `group-exports` and `exports-last`, is this something we actually want? I tend to lean in favor, but it requires a non-trivial amount of work, and I wouldn't consider it a no-brainer.
- `no-restricted-paths`, from what I understand this is only useful in the context where "restricted zones" are defined, but I don't think we have those

I've ran into situations, on other repos, where an excessive amount of resource-intensive linting rules led to performance issues for some contributors, hence my preoccupation with ensuring that additional rules serve a useful purpose.

Tagging the rest of the team in case they'd like to voice an opinion on that. @holgerd77 @acolytec3 @jochem-brouwer @ScottyPoi @g11tech

Reply:
I don't have past experience with complex linting setups beyond the monorepo so can't comment on performance, but I don't notice any differences between running the linter on master vs this PR on my VSCode setup.

As far as individual rules in this set go, the only one we'll have to revisit is the `import/extensions` if/when we add ESM support.  Otherwise, this list looks good to me and I like having all the imports organized.

Reply:
Thanks guys for the additional comments! Looks great @faustbrian, I'm on board with the remaining eslint rules! Good to know that there are solutions if we ever run into performance issues.